src/components/Swap/Swap.tsx

```diff
@@ -61,6 +64,8 @@ export interface Pools {
 }
 
 export interface ISwap {
+  tokenFrom: PublicKey | null
```

Comment:
Is there a need to pass these props? What about `tokens[tokenFromIndex].address`?